src/Reader/Dynamic.hs

```diff
@@ -21,11 +28,19 @@ data Reader r es = MkReader
   { askImpl :: Eff es r,
     localImpl ::
       forall e a.
-        (r -> r) ->
-        (Reader r e -> Eff es a) ->
-        Eff (e :& es) a
+      (r -> r) ->
+      (forall e1. Reader r e1 -> Eff (e1 :& e) a) ->
+      Eff (e :& es) a
```

Comment:
It's hard to know this is the correct type for `localImpl`. I know it because I've built up an intuition for these things whilst playing with them, but I'm not sure how to communicate the intuition to others. I suppose a good check is "can you write a `Handle` instance for it?".

Reply:
Indeed. Tbh, my approach so far has been mainly looking at examples in the `bluefin` repo/source and playing type tetris. I do not have the intuition yet, which I'm sure shows :joy:.

> I suppose a good check is "can you write a Handle instance for it?".

If writing a `Handle` instance (positively) constrains the types, then I think that would be a great addition. As I mentioned in [the other PR](https://github.com/tbidne/bluefin-higher-order/pull/1#issuecomment-2480852039), I missed the importance of `Handle`.

Reply:
> my approach so far has been mainly looking at examples in the bluefin repo/source and playing type tetris. I do not have the intuition yet, which I'm sure shows 😂.

You're also the first person to try to do these things systematically so you are breaking new ground! I myself hadn't even tried it.